### ISSUE TITLE
Option to prefetch videos before starting a stream.

### ISSUE THREAD

**Issue by IHateTheProgrammingClass** [TeamNewPipe/NewPipe#6759]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

There could be an option in the settings that enables automatically prefetching a video if autoplay is not enabled.

So if autoplay is enabled:

Video downloads, and autoplays

If autoplay disabled but "prefetch" is enabled:

Video downloads, but no autoplay

If prefetching is disabled:

No download, only start when tapping play, no autoplay

Optionally the feture can be dependent on what type of connection is present. So it could for example be set to only do this while on WiFi.

Maybe this could somehow be connected to autoplay itself? As an extra toggle under autoplay could appear (if it's disabled) that would enable prefetching while bot autoplaying. This could benefit people who although do not use autoplay ever, but would like a faster video streaming experience on WiFi.

#### How will you/everyone benefit from this feature?

Faster video streaming, more versatility.

**Comment by SameenAhnaf** [comment]
This feature is especially useful if someone wants to read description and comments for a long time and start video later on a high latency Wi-Fi.

**Comment by mhmdanas** [comment]
I think this is a duplicate of #4932. @IHateTheProgrammingClass feel free to post any thoughts you have there.
